### PR TITLE
Fix #208, fix #209, see changelog below.

### LINKED ISSUE #208: Change ending character for custom commands
Now the ending character for custom commands is `!`, but the same character is used as a header by the minor servos protocol. In order to implement the minor servos simulator properly and to avoid issues it is mandatory to replace the `!` character for custom commands. The new character should not be a character that starts or ends any command from any known protocol.

### LINKED ISSUE #209: `test_server.py` fails when `ACSDATA` environment variable is set
The path for the log file from which the tests tries to read some entries points to the wrong directory. It should take into account the presence of absence of the `ACSDATA` environment variable.

### PR DESCRIPTION
**Pull request by giuseppe-carboni**
Issue #208, replaced ending character for custom commands in order to avoid clashes with the starting character of the minor servos protocol

Issue #209, now the `test_server.py` test file accounts for the presence of the `ACSDATA` environment variable and points to the correct path for some tests

Also, documentation for `common.py` and `server.py` has been improved.

### CODE REVIEW

File: simulators/common.py [hunk 1]
```diff
@@ -2,36 +2,54 @@
...
+            client) when the message is completed.
+        :rtype: boolean, string
+        :raise ValueError: when the declared length of the message exceeds the
+            maximum expected length, when the sent message carries a wrong
+            checksum or when the client asks to execute an unknown command."""
 
 
 class SendingSystem(BaseSystem):
+    """This class implements a server that, as soon as a client(s) opens a
+    connection, it starts periodically sending to the said client(s)
```
Comment (marco-buttu) [inline simulators/common.py]: connection, starts (without it). Actually the sentence is not clear to me. Maybe: *Implement a server that sends some system information data to every connected client.*

File: simulators/server.py [hunk 2]
```diff
@@ -21,15 +21,16 @@
...
 class BaseHandler(BaseRequestHandler):
     """This is the base handler class from which `ListenHandler` and
-    `SendHandler` classes are inherited. It only defines the custom
-    header and tail for accepting some commands not related to the system
-    protocol, and the `_execute_custom_command` method to parse the received
-    custom command.
-    I.e. a $system_stop! command will stop the server, a $error! command
-    will configure the system in order to respond with errors, etc.
-    """
+    `SendHandler` classes are inherited. It only defines the custom header
```
Comment (marco-buttu) [inline simulators/server.py]: First sentence is plural, second is singular

File: simulators/server.py [hunk 3]
```diff
@@ -64,10 +65,20 @@ def _execute_custom_command(self, msg_body):
...
     def setup(self):
+        """This method gets called as soon as a client connects to the server.
+        """
         logging.info('Got connection from %s', self.client_address)
         self.custom_msg = b''
 
     def handle(self):
+        """This method gets called right after the `setup` method ends its
+        execution. It handles incoming messages, whether they are received via
+        a TCP or a UDP socket. It pass down the the `System` class the received
```
Comment (marco-buttu) [inline simulators/server.py]: It passes... the the

File: simulators/server.py [hunk 4]
```diff
@@ -64,10 +65,20 @@ def _execute_custom_command(self, msg_body):
...
+        messages one byte at a time in order for the `System.parse()` method to
+        work properly. It then returns the `System` response when necessary.
+        It also constantly listens for custom commands that does not belong to
```
Comment (marco-buttu) [inline simulators/server.py]: do not

File: simulators/server.py [hunk 5]
```diff
@@ -120,9 +131,15 @@ def _handle(self, msg):
...
 
     def setup(self):
+        """This method gets called whenever a client connects to the server."""
         logging.info('Got connection from %s', self.client_address)
 
     def handle(self):
+        """This method gets called right after the `setup` method ends its
+        execution. It handles messages that the server has to periodically send
+        to its connected client(s). It also constantly listens for custom
+        commands that does not belong to a specific `System` class, but are
```
Comment (marco-buttu) [inline simulators/server.py]: do not

**Comment by giuseppe-carboni** [comment]
> In addition to all previous comments, a comment about docstrings. As you can see here: https://www.python.org/dev/peps/pep-0257, they do not have to describe the object. For instance:
> 
> ```
> def add(a, b):
>     "Add two objects"
>     return a + b
> ```
> 
> and not:
> 
> ```
> def add(a, b):
>     "This function returns two objects"
>     return a + b
> ```
> 
> If you wrote all docstrings in the second way, left them as they are because it is too much work to change everything and also because changing them we can add typos and mistakes. So, we just keep in mind this note for future projects...

Unfortunately I always took the second approach. I will avoid describing functions that way on future modules.

File: simulators/common.py [hunk 6]
```diff
@@ -2,36 +2,54 @@
...
-        """This method sends back to the server the message `$server_shutdown!`
+        """This method sends back to the server the message `$server_shutdown%`
         ordering it to stop accepting requests and to shut down."""
-        return '$server_shutdown!'
+        return '$server_shutdown%'
 
 
 class ListeningSystem(BaseSystem):
+    """This class implements a server that wait for its client(s) to send a
+    command, it can then answer back when required. Unlike the `SendingSystem`
```
[the remark this answers is not in the document]
Comment (giuseppe-carboni) [reply simulators/common.py]: The way it's written is already correct, the verb is `can`, not `answer`.

File: simulators/server.py (hunk 2 above)
Comment (giuseppe-carboni) [reply simulators/server.py]: The second sentence starts with `It`, that refers to the `BaseHandler` class only, not the other two mentioned classes.

**Pull request merged**